src/ngamsPlugIns/ngamsPlugIns/ngamsSdmMultipart.py

```diff
@@ -37,189 +37,169 @@
+# UID_EXPRESSION = re.compile("[xX][0-9,a-f]{3,}/[xX][0-9,a-f]{1,}$")
+
+# This matches the new UID structure uid://X1/X2/X3#kdgflf
+# UID_EXPRESSION = re.compile("^[uU][iI][dD]:/(/[xX][0-9,a-f,A-F]+){3}(#\w{1,}|/\w{0,}){0,}$")
+
+# Update for the new assignment of archiveIds (backwards compatible)
+UID_EXPRESSION = re.compile(r"^[uU][iI][dD]://[aAbBcCzZxX][0-9,a-z,A-Z]+(/[xX][0-9,a-z,A-Z]+){2}(#\w{1,}|/\w{0,}){0,}$")
 
-def specificTreatment(fo):
+
+def specific_treatment(file_path):
     """
     Method contains the specific treatment of the file passed from NG/AMS.
 
-    fo:         File object
+    file_path:  File path
 
     Returns:    (file_id, finalFileName, type);
                 The finalFileName is a string containing the name of the final
                 file without extension. type is the mime-type from the header.
     """
-    import rfc822, cgi, re
-    _EXT = '.msg'
-
-    filename = fo.name
-
-#    uidTempl = re.compile("[xX][0-9,a-f]{3,}/[xX][0-9,a-f]{1,}$")
-# This matches the old UID of type    X0123456789abcdef/X01234567
-
-#    uidTempl = re.compile("^[uU][iI][dD]:/(/[xX][0-9,a-f,A-F]+){3}(#\w{1,}|/\w{0,}){0,}$")
-# This matches the new UID structure uid://X1/X2/X3#kdgflf
-
-    # update for the new assignment of archiveIds (backwards compatible)
-    uidTempl = re.compile("^[uU][iI][dD]://[aAbBcCzZxX][0-9,a-z,A-Z]+(/[xX][0-9,a-z,A-Z]+){2}(#\w{1,}|/\w{0,}){0,}$")
-
-    try:
-        message = rfc822.Message(fo)
-        type, tparams = cgi.parse_header(message["Content-Type"])
-    except Exception as e:
-        err = "Parsing of mime message failed: " + str(e)
-        errMsg = genLog("NGAMS_ER_DAPI_BAD_FILE", [os.path.basename(filename),
-                                                   _PLUGIN_ID, err])
-        raise Exception(errMsg)
+    filename = os.path.basename(file_path)
     try:
-        # CAUTION!!! parse_header returns stuff in lower case that's why it is not used here
-        almaUid = message["alma-uid"]
-    except:
-        try:
-            almaUid = message["Content-Location"]
-        except:
-            err = "Mandatory alma-uid or Content-Location parameter not found in mime header!"
-            errMsg = genLog("NGAMS_ER_DAPI_BAD_FILE", [os.path.basename(filename),
-                                               _PLUGIN_ID, err])
-            raise Exception(errMsg)
-
-    if not uidTempl.match(almaUid):
-        err = "Invalid alma-uid found in Content-Location: " + almaUid
-        errMsg = genLog("NGAMS_ER_DAPI_BAD_FILE", [os.path.basename(filename),
-                                               _PLUGIN_ID, err])
-        raise Exception(errMsg)
-
-# Now, build final filename. We do that by looking for the UID in
-# the message mime-header.
-
-# The final filename is built as follows: <almaUidR>.<ext>
-# where almaUidR has the slash character in the UID replaced by colons.
+        with open(file_path, 'rb') as file_object:
+            message = email.message_from_file(file_object)
```

Comment:
This works in python 2 but not in 3, as `message_from_file` needs a file object opened in text mode instead of binary.

You *need* to open the file in binary mode though (to avoid decoding, which would otherwise fail), so you'll need to do something like:

```python
# at the module level, so it happens only once
message_from_file = email.message_from_file
if sys.version_info[0] > 2:
    message_from_file = email.message_from_binary_file
```

and then

```diff
-            message = email.message_from_file(file_object)
+            message = message_from_file(file_object)
```

Reply:
I have applied the change you suggested. Many thanks for solving the issue.